src/Servers/Kestrel/Core/src/Adapter/Internal/AdaptedPipeline.cs

```diff
@@ -145,7 +154,16 @@ private async Task ReadInputAsync(Stream stream)
                 // The application could have ended the input pipe so complete
                 // the transport pipe as well
                 _transport.Input.Complete();
+
+                // Cancel any pending reads from the application
+                Output.Reader.CancelPendingRead();
```

Comment:
You just completed the input writer. What point does clearing the output reader have? Was this supposed to be CancelPendingWrite?

Reply:
I think this could result in truncated output.